indexing-service/src/main/java/io/druid/indexing/common/TaskLock.java

```diff
@@ -33,19 +33,25 @@
   private final String dataSource;
   private final Interval interval;
   private final String version;
+  private final Integer priority;
+  private final Boolean upgraded;
 
   @JsonCreator
   public TaskLock(
       @JsonProperty("groupId") String groupId,
       @JsonProperty("dataSource") String dataSource,
       @JsonProperty("interval") Interval interval,
-      @JsonProperty("version") String version
+      @JsonProperty("version") String version,
+      @JsonProperty("priority") Integer priority,
+      @JsonProperty("upgraded") Boolean upgraded
   )
   {
     this.groupId = groupId;
     this.dataSource = dataSource;
     this.interval = interval;
     this.version = version;
+    this.priority = priority;
+    this.upgraded = upgraded == null ? false : upgraded;
```

Comment:
I just did it for consistency reasons to keep `priority` and `upgraded` objects instead of primitive types. Are there any benefits in keeping them as primitive type ?